src/redux/reducers/todoReducer.js

```diff
@@ -0,0 +1,56 @@
+import * as types from '../types';
+
+const initialState = {
+    todosData: [],
+};
+
+const saveToLocalStorage = function (state) {
+    localStorage.setItem('todos', JSON.stringify(state));
+}
+
+const retrieveFromLocalStorage = function () {
+    return JSON.parse(localStorage.getItem('todos'))
+}
+
+export default function (state = initialState, action) {
+    let newState;
+    switch (action.type) {
+        case types.FETCH_TODOS:
+            const retrievedState = retrieveFromLocalStorage() || initialState
```

Comment:
This action is not necessary at all. you could directly retrieved data from local storage and saved it into the initial state with a fallback.